src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,124 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+
+    private Object[] elements;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growIfFull();
+        elements[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
+        if (checkAddIndex(index)) {
+            growIfFull();
+            elements = addResize(value, index);
+            size++;
+        }
+    }
 
+    private Object[] addResize(T value, int index) {
+        Object[] update = new Object[elements.length];
+        System.arraycopy(elements, 0, update, 0, index);
+        update[index] = value;
+        System.arraycopy(elements, index, update, index + 1,
+                size - index);
+        return update;
```

Comment:
Why we need new array for this operation. We can copy in the same array, and don't need return update
```suggestion
    private void addResize(T value, int index) {
        System.arraycopy(elements, index, update, index + 1,
                size - index);
        elements[index] = value;
```